packages/contracts/contracts/attester-utils/ReimburseAttestation.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import {Unirep} from '../Unirep.sol';
+
+/**
+ * @title ReimburseAttestation
+ * @dev An attester can inherit this contract to hold budget and allow the callers of the contract (attester app users) to be reimbersed for attestations.
+ 	This would make the publicly funding operation of attesters more modular and transparent.
+	e.g. An attester could directly get funded from a gitcoin round for its operation.
+ */
+contract ReimburseAttestation {
+    bool public acceptDonations;
+    bool public acceptReReimburseUST;
+
+    mapping(address => bool) public whitelist;
+
+    Unirep public unirep;
+
+    // id of the attester this budget contract is deployed for
+    uint160 public attesterId;
+
+    enum TxType {
+        Signup,
+        UST
+    }
+    TxType public txType;
+
+    event Reimbursed(address indexed recipient, uint amount, TxType txType);
+    event FundsReceived(address sender, uint amount);
+
+    constructor(Unirep _unirep, uint160 _attesterId) {
+        unirep = _unirep;
+        attesterId = _attesterId;
+        acceptDonations = true;
+        acceptReReimburseUST = true;
+    }
+
+    /**
+     * @dev Add users to whitelist
+     */
+    function addToWhitelist(
+        address[] calldata addresses,
+        address owner
+    ) public virtual {
+        require(
+            msg.sender == owner,
```

Comment:
I'm not sure what this check is. It's checking for the sender to be an argument to the function, was this meant to be something else? Like storing `owner` as a state variable?